Assets/Scripts/Enemy/EnemyController.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Bullet;
+
+namespace Enemy
+{
+    public class EnemyController
+    {
+		
+        private BulletController bulletController;
+        public EnemyController(TankModel model, EnemyView prefab)
+        {
+            tankModel = model;
+            enemyView = GameObject.Instantiate<EnemyView>(prefab);
+
+            //X location where new tank is spawned
+            Debug.Log("Tank type -" + model.tankType);
+            enemyView.Init(this);
+            bulletController = BulletService.Instance.GetBulletController(this);
```

Comment:
why are you getting a bullet controller in the constructor of the enemy controller?

Reply:
you should get the bullet controller when you have to fire the bullet not before it